stdlib/src/collections/string.mojo

```diff
@@ -29,88 +29,189 @@ from sys.intrinsics import _type_is_eq
 from hashlib._hasher import _HashableWithHasher, _Hasher
 
 from utils import (
-    Span,
     IndexList,
     StringRef,
-    StringSlice,
     Variant,
     Writable,
     Writer,
 )
+from utils.span import Span, AsBytes
 from utils.string_slice import (
+    StringSlice,
     _utf8_byte_type,
     _StringSliceIter,
     _unicode_codepoint_utf8_byte_length,
     _shift_unicode_to_utf8,
     _FormatCurlyEntry,
     _CurlyEntryFormattable,
     _to_string_list,
+    Stringlike,
+    _utf8_first_byte_sequence_length,
+    _is_continuation_byte,
+    _is_valid_utf8,
 )
+from builtin.builtin_list import _lit_mut_cast
 
 # ===----------------------------------------------------------------------=== #
 # ord
 # ===----------------------------------------------------------------------=== #
 
 
-fn ord(s: String) -> Int:
-    """Returns an integer that represents the given one-character string.
+fn ord[T: Stringlike, //](ref [_]s: T) -> Int:
+    """Returns the unicode codepoint for the character.
 
-    Given a string representing one character, return an integer
-    representing the code point of that character. For example, `ord("a")`
-    returns the integer `97`. This is the inverse of the `chr()` function.
+    Parameters:
+        T: The Stringlike type.
 
     Args:
         s: The input string slice, which must contain only a single character.
 
     Returns:
         An integer representing the code point of the given character.
-    """
-    return ord(s.as_string_slice())
 
+    Examples:
+    ```mojo
+    print(ord("a"), ord("€")) # 97 8364
+    ```
+    .
+    """
 
-fn ord(s: StringSlice) -> Int:
-    """Returns an integer that represents the given one-character string.
+    # FIXME(#933): llvm intrinsic can't recognize !pop.scalar<ui8> value when
+    # trying to fold ctlz at comp time
```

Comment:
Sorry to ping you @lattner, but issue #933 is popping up again and I had to make this ugly workaround :cry: